ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualSignedTransaction.kt

```diff
@@ -47,16 +48,46 @@ interface ConsensualSignedTransaction {
      *
      * @param publicKey The private counterpart of the specified public key will be used for signing the
      *      [ConsensualSignedTransaction].
-     * @return Returns a new [ConsensualSignedTransaction] containing the applied signature.
+     * @return Returns the new [ConsensualSignedTransaction] containing the applied signature and the signature itself.
      */
-    fun addSignature(publicKey: PublicKey): ConsensualSignedTransaction
+    @Suspendable
+    fun addSignature(publicKey: PublicKey): Pair<ConsensualSignedTransaction, DigitalSignatureAndMetadata>
+
+    /**
+     * Adds a signature to the current [ConsensualSignedTransaction].
+     *
+     * @param signature The signature to be added to the [ConsensualSignedTransaction].
+     *
+     * @return Returns a new [ConsensualSignedTransaction] containing the new signature.
+     */
+    fun addSignature(signature: DigitalSignatureAndMetadata): ConsensualSignedTransaction
+
+
+    /**
+     * Crosschecks the missing signatures with the available keys and signs the transaction with their intersection
+     * if there are any. (Disabled until crypto support becomes available.)
+     *
+     * @return Returns a new [ConsensualSignedTransaction] containing the new signature.
+     *
+    @Suspendable
+     * @return Returns the new [ConsensualSignedTransaction] containing the applied signature and a
+     *          list of added signatures.
+    fun addMissingSignatures(): Pair(ConsensualSignedTransaction, list<DigitalSignatureAndMetadata>)
+    */
 
     /**
      * Gets the signing keys for any missing transaction signatures.
      *
      * @return Returns a [Set] of [PublicKey] representing the signing keys for any missing transaction signatures.
      */
     fun getMissingSigningKeys(): Set<PublicKey>
+
+    /**
+     * Verify all available signatures and whether there are any missing ones.
+     *
+     * @throws TransactionVerificationException in both cases.
```

Comment:
That comment in ambiguous - I am guessing "both cases" means either a signature not verifying or a signature missing?

Reply:
Yes. Fixed.